app/backend/src/couchers/notifications/push.py

```diff
@@ -10,29 +13,50 @@
 from couchers.proto.internal import jobs_pb2
 
 
+@dataclass(frozen=True, slots=True)
+class PushNotificationContent:
+    """Defines the user-visible content of a push notification."""
+
+    # Android reference: https://developer.android.com/develop/ui/views/notifications
+    # iOS reference: https://developer.apple.com/documentation/usernotifications/unnotificationcontent
+
+    MAX_TITLE_LENGTH: ClassVar[int] = 500
+    MAX_BODY_LENGTH: ClassVar[int] = 2000
+
+    title: str
+    """A localized title for the notification, this should be a very short string (2-4 words)."""
+    body: str
+    """The localized text of the notification body."""
+    action_url: str | None = None
+    """The URL to open when the notification is clicked. If None, will open the app's main URL."""
+    icon_url: str | None = None
+    """A URL to the icon to show in the notification. If None, will use the default app icon."""
+
+
 def push_to_subscription(
     session: Session,
     *,
     push_notification_subscription_id: int,
     user_id: int,
     topic_action: str,
+    content: PushNotificationContent,
     key: str | None = None,
-    title: str = "",
-    body: str = "",
-    icon: str | None = None,
-    url: str | None = None,
     ttl: int = 0,
 ) -> None:
+    title = config["NOTIFICATION_PREFIX"] + content.title[: PushNotificationContent.MAX_TITLE_LENGTH]
```

Comment:
What do you think about raising an error here? Such a long title seems more like a bug. Or maybe send an event to sentry, but still proceed with sending the notification